File: ios/Event/Event.swift
//
//  Event.swift
//  react-native-vonage-voice
//
//  Created by Volodymyr Smolianinov on 03/08/2024.
//

import Foundation

enum Event: String {
    case callEvents
    case register
    case voipTokenInvalidated
    case audioRouteChanged
    case muteChanged
    
    static var supportedEvents: [String] {
        return [
            Event.callEvents.rawValue,
            Event.register.rawValue,
            Event.voipTokenInvalidated.rawValue,
            Event.audioRouteChanged.rawValue,
            Event.muteChanged.rawValue,
        ]
    }
}

File: android/src/main/java/com/vonagevoice/event/Event.kt
package com.vonagevoice.event

enum class Event(val value: String) {
    CALL_EVENTS("callEvents"),
    REGISTER("register"),
    VOIP_TOKEN_INVALIDATED("voipTokenInvalidated"),
    AUDIO_ROUTE_CHANGED("audioRouteChanged"),
    MUTE_CHANGED("muteChanged");

    companion object {
        fun supportedEvents(): List<String> = values().map { it.value }
    }
}
